File: AlgDat/oving1/Sprengstoff.java
import java.io.*;
import java.util.*;

public class Sprengstoff {

    private static int spor(Kubbe kubbe) {
	int max = kubbe.vekt;
	Kubbe next = kubbe.neste;
	while(next != null){
		if(next.vekt>max){
			max = next.vekt;
		}
		next = next.neste;
	}
	return max;
    }

    public static void main(String args[]) {
        BufferedReader in;
        if (args.length > 0) {
            try {
              in = new BufferedReader(new FileReader(args[0]));
            }
            catch (FileNotFoundException e) {
                System.out.println("Kunne ikke Œpne filen " + args[0]);
                return;
            }
        }
        else {
            in = new BufferedReader(new InputStreamReader(System.in));
        }
        try {
            Kubbe forste = null, siste = null, forrige_siste = null;
            String linje = in.readLine();
            while (linje != null) {
            	forrige_siste = siste;
                int i = Integer.parseInt(linje);
                siste = new Kubbe(i);
                	if(forste == null){
                		forste = siste;
                	}
                	else{
                		forrige_siste.neste = siste;
                	}
                linje = in.readLine();
            }
            System.out.println(spor(forste));
        }
        catch (Exception e) {
            e.printStackTrace();
        }
    }
}

class Kubbe {
	int vekt;
	Kubbe neste;
	
	public Kubbe(int vekt) { 
		this.vekt = vekt;
	}
}

File: AlgDat/oving1/sprengstoff.py
from sys import stdin

class Kubbe:
    vekt = None
    neste = None
    def __init__(self, vekt):
        self.vekt = vekt 
        self.neste = None 

def spor(kubbe):
    
    max = kubbe.vekt
    next = kubbe.neste
    while next != None:
        if (next.vekt > max):
            max = next.vekt
        next = next.neste
    return max


# Oppretter lenket liste
forste = None
siste = None
for linje in stdin:
    forrige_siste = siste
    siste = Kubbe(int(linje))
    if forste == None:
        forste = siste
    else:
        forrige_siste.neste = siste

# Kaller loesningsfunksjonen og skriver ut resultatet
print spor(forste)
